cdap-app-fabric/src/main/java/io/cdap/cdap/config/PreferencesTable.java

```diff
@@ -289,6 +313,26 @@ private void checkSeqId(String namespace, String type, String name, long seq)
     return Collections.emptyMap();
   }
 
+  /**
+   * Get the metadata of preferences
+   *
+   * @param namespace namespace of the entity
+   * @param type      preferences for what kind of entity (e.g. instance, namespace, application or program)
+   * @param name      the name of the entity (e.g. name of the application)
+   * @return metadata of preferences, Null if preferences for the entity are not found.
+   * @throws IOException if unable to read the data from underlying storage
+   */
+  @Nullable
+  private PreferencesMetadata getMetadata(String namespace, String type, String name) throws IOException {
+    List<Field<?>> primaryKey = getPrimaryKey(namespace, type, name);
+    Optional<StructuredRow> row = table.read(primaryKey);
+    if (!row.isPresent()) {
+      return null;
+    }
+    Long seqId = row.get().getLong(StoreDefinition.PreferencesStore.SEQUENCE_ID_FIELD);
+    return new PreferencesMetadata((seqId == null ? null : seqId.longValue()));
```

Comment:
Done. 

Looks to me mostly likely this would never return null because of the row.isPresent check. But the API contract says row.get().getLong() can, so I think we should handle that case.

Reply:
It makes sense for the Row contract, but I'm not sure it makes sense in the PreferencesTable contract to return an empty PreferencesMetadata object, as it means the underlying table is in an invalid state? 

Practically speaking I don't think it matters much, as I think it can only happen if somebody manually modifies the underlying table, but it's probably better to throw an IllegalStateException if it is null. This would allow PreferenceMetadata to use a long instead of Long, and users of that class to not need to worry about nulls.